cmd/collectors/rest/plugins/snapmirror/snapmirror.go

```diff
@@ -47,28 +51,60 @@ func (my *SnapMirror) Init() error {
 
 	my.query = "api/private/cli/volume"
 	my.svmVolToNode = make(map[string]string)
+	my.svmPeerDataMap = make(map[string]string)
+
+	my.data = matrix.New(my.Parent+".SnapMirror", "snapmirror", "snapmirror")
+
+	exportOptions := node.NewS("export_options")
+	instanceLabels := exportOptions.NewChildS("instance_labels", "")
+	instanceKeys := exportOptions.NewChildS("instance_keys", "")
+
+	if exportOption := my.ParentParams.GetChildS("export_options"); exportOption != nil {
+		if exportedLabels := exportOption.GetChildS("instance_labels"); exportedLabels != nil {
+			for _, l := range exportedLabels.GetAllChildContentS() {
+				instanceLabels.NewChildS("", l)
+			}
+		}
+		if exportedLabels := exportOption.GetChildS("instance_keys"); exportedLabels != nil {
+			for _, k := range exportedLabels.GetAllChildContentS() {
+				instanceKeys.NewChildS("", k)
+			}
+		}
+	}
+	my.data.SetExportOptions(exportOptions)
 
 	// Assigned the value to nodeUpdCounter so that plugin would be invoked first time to populate cache.
 	my.nodeUpdCounter = PluginInvocationRate
-
 	return nil
 }
 
 func (my *SnapMirror) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
+	// Purge and reset data
+	my.data.PurgeInstances()
+	my.data.Reset()
+
+	// Set all global labels from Rest.go if already not exist
+	my.data.SetGlobalLabels(data.GetGlobalLabels())
 
 	if my.nodeUpdCounter >= PluginInvocationRate {
 		my.nodeUpdCounter = 0
 		if err := my.updateNodeCache(); err != nil {
 			return nil, err
 		}
 		my.Logger.Debug().Msg("updated node cache")
+
+		cluster, _ := data.GetGlobalLabels().GetHas("cluster")
```

Comment:
should the `exists` return value be used before calling `getSVMPeerData`?

Reply:
yes.